packages/hub/indexing/document-context.js

```diff
@@ -150,67 +166,112 @@ module.exports = class DocumentContext {
     searchDocOut[field.id] = related;
   }
 
-  async _build(type, id, jsonapiDoc, searchTree, depth) {
-    let contentType = this.schema.types.get(type);
-    if (!contentType) {
-      log.warn("ignoring unknown document type=%s id=%s", type, id);
-      return;
-    }
+  _buildSearchTree(searchTree, contentType, segments) {
+    if (!segments.length) { return contentType; }
+    if (searchTree[segments[0]]) { return; }
+
+    let field = contentType.realAndComputedFields.get(segments[0]);
+    if (!field || !field.relatedTypes) { return contentType; }
+
+    let relatedTypes = Object.keys(field.relatedTypes);
+    if (!relatedTypes.length) { return contentType; }
 
+    //TODO how to decide which related type to use?
+    searchTree[segments[0]] = this._buildSearchTree(Object.assign({}, searchTree), this.schema.types.get(relatedTypes[0]), segments.slice(1));
```

Comment:
@ef4 I have a question. when using the schema to look up the type for a field, i see that for fields they use a `relatedTypes` that is an object whose keys are the related type for the field. This means that it is possible for a field to be related to more than one type. I'm unsure what that semantically means when a field is associated with more than one type. I just arbitrarily chose a related type, but I have a feeling that because I don't understand this structure, I probably did the wrong thing. Can you help me to understand the `Field.relatedTypes` better? Maybe my approach to deriving the type for a field is incorrect?

Reply:
Yeah, the semantics here are not fully fleshed out.

In practice right now, practical relationship fields should have one content-type in related-types. The schema is more flexible than that, with the idea that it could eventually support heterogeneous relationships, but that doesn't really get used anywhere and it would take more work to make it do anything. 

So you sense of things being off is correct, but I think OK for the present.